libs/stocks/feature-shell/src/lib/stocks/stocks.component.ts

```diff
@@ -25,19 +29,99 @@ export class StocksComponent implements OnInit {
     { viewValue: 'One month', value: '1m' }
   ];
 
+  get shouldDisplayDateError(): boolean {
+    return (
+      this.stockPickerForm.errors && this.stockPickerForm.errors.fromAfterTo
+    );
+  }
+
   constructor(private fb: FormBuilder, private priceQuery: PriceQueryFacade) {
-    this.stockPickerForm = fb.group({
-      symbol: [null, Validators.required],
-      period: [null, Validators.required]
-    });
+    const today = new Date();
+    const year = today.getFullYear();
+    const month = today.getMonth();
+    const date = today.getDate();
+    // according to API documentation the oldest records will be 15 years old
+    // https://iexcloud.io/docs/api/#charts
+    const maxAgeOfAPIData = 15;
+
+    this.maxDate = new Date(year, month, date);
+    this.minDate = new Date(year - maxAgeOfAPIData, month, date);
+
+    this.stockPickerForm = fb.group(
+      {
+        symbol: [null, Validators.required],
+        to: [today, [Validators.required]],
+        from: [today, [Validators.required]]
+      },
+      { validators: DateRangeValidator }
+    );
+
+    this.stockPickerForm.valueChanges
+      .pipe(
+        debounceTime(500),
+        filter(() => this.stockPickerForm.invalid),
+        map(() => {
+          if (this.shouldDisplayDateError) {
```

Comment:
Not a fan of this feature request, would push back because in my opinion it is more useful to the user to correct their mistake themselves (if clearly shown on UI)